### PR TITLE
Unique validation

### PR DESCRIPTION
**Pull request by SteRiccio**
## Description

Attributes that are multiple or children of root entity (cluster) or children of a multiple entity (tree) can be marked as unique.
When marked as unique, duplicate values will be reported as errors in the data entry UI and in the validation report. 

- #### Added _Unique_ checkbox in Node Def editor / Validations
   
Screenshot_2021-05-05_16-31-07

## Types of changes

- [ ] Docs change
- [ ] Refactoring
- [ ] Dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How has this been tested

The E2E test of the validation report has been modified to test duplicate values in attributes marked as unique.

#### Do you consider this PR needs further testing?

- [x] No
- [ ] Yes

## UI changes

Screenshot_2021-05-05_17-08-41
Screenshot_2021-05-05_17-09-30

### CODE REVIEW

**Comment by minotogna** [comment]
Task linked: [CU-g13mf6 Unique validation](https://app.clickup.com/t/g13mf6)

File: core/i18n/resources/en.js [hunk 1]
```diff
@@ -762,6 +763,7 @@ $t(common.cantUndoWarning)`,
       entityKeyDuplicate: 'Duplicate entity key',
       nodesMaxCountExceeded: '{{nodeDefName}} nodes must be less than or equal to {{maxCount}}',
       nodesMinCountNotReached: '{{nodeDefName}} nodes must be more than or equal to {{minCount}}',
+      uniqueAttributeDuplicate: 'Duplicate unique attribute',
```
Comment (minotogna) [inline core/i18n/resources/en.js:766]: better to say Duplicate attribute?

Comment (SteRiccio) [reply core/i18n/resources/en.js:766]: or maybe just "Duplicate value"

Comment (minotogna) [reply core/i18n/resources/en.js:766]: sounds good !

File: core/record/_record/recordReader.js [hunk 2]
```diff
@@ -201,15 +199,56 @@ export const getParentCodeAttribute = (survey, parentNode, nodeDef) => record =>
...
+      ...siblingsAcc,
+      ...getNodeChildrenByDefUuid(siblingEntity, NodeDef.getUuid(attributeDef))(record),
+    ],
+    []
+  )
+}
+
+const _isNodeDefUnique = (nodeDef) => NodeDefValidations.isUnique(NodeDef.getValidations(nodeDef))
+
+export const getAttributesUniqueDependent = ({ survey, record, node }) => {
```
Comment (ramirobg94) [inline core/record/_record/recordReader.js:231]: do we need to add byId or something like that?

**Review by ramirobg94 — APPROVED** [review]

**Review by minotogna — APPROVED** [review]

File: core/record/_record/recordReader.js (hunk 2 above)
Comment (SteRiccio) [reply core/record/_record/recordReader.js:231]: yes you are right, a name like getUniqueDependentsByUuid would be better... but we have so many similar cases in the code. The return type will be more clear in a near future using TypeScript.

**Review by minotogna — APPROVED** [review]

**Review by minotogna — DISMISSED** [review]

**Review by ramirobg94 — APPROVED** [review]

**Pull request merged**